client/notebooks/Demo_Client_Notebook_Smartnoise-SQL.ipynb

```diff
@@ -800,415 +622,401 @@
    "source": [
-    "Dr. Antartica has now a differentially private estimation of the bill length of all birds and is confident to use the library for the rest of her analyses."
+    "She can now do all the postprocessing that she wants with the returned data without adding any privacy risk. "
    ]
   },
   {
-   "cell_type": "markdown",
-   "id": "04929993",
+   "cell_type": "code",
+   "execution_count": 23,
+   "id": "f72b19d0",
    "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Standard error of bill length: 0.16.\n"
+     ]
+    }
+   ],
    "source": [
-    "## Step 4: Penguin statistics"
+    "# Get standard error\n",
+    "standard_error = std_bill_length/np.sqrt(nb_penguin)\n",
+    "print(f\"Standard error of bill length: {np.round(standard_error, 2)}.\")"
    ]
   },
   {
-   "cell_type": "markdown",
-   "id": "bbbca191",
+   "cell_type": "code",
+   "execution_count": 24,
+   "id": "62630a03",
    "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "The 95% confidence interval of the bill length of all penguins is [42.98, 43.61].\n"
+     ]
+    }
+   ],
    "source": [
-    "### Confidence intervals for flipper length over the whole population"
+    " # Compute the 95% confidence interval\n",
+    "ZSCORE = 1.96\n",
+    "lower_bound, upper_bound = avg_bill_length - ZSCORE*standard_error, avg_bill_length + ZSCORE*standard_error\n",
+    "print(f\"The 95% confidence interval of the bill length of all penguins is [{np.round(lower_bound, 2)}, {np.round(upper_bound, 2)}].\")"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "9d41bd58",
+   "id": "26d04824-ff41-4d25-8a4e-1506a416dd0b",
    "metadata": {},
    "source": [
-    "She is first interested to have a better idea of the distribution of flipper length of all species. She already has the mean from step 3, so she just need to compute the standard deviation and know the number of penguins in the dataset. As it is just an exploration step, she uses very little budget values."
+    "## Note on budget with Smartnoise-SQL (Advanced)"
    ]
   },
   {
-   "cell_type": "code",
-   "execution_count": 23,
-   "id": "04b376ef",
+   "cell_type": "markdown",
+   "id": "c9aa0b56-bda3-405e-9f33-ae7135dbfeba",
    "metadata": {},
-   "outputs": [],
    "source": [
-    "QUERY = \"SELECT COUNT(bill_length_mm) AS nb_penguin, STD(bill_length_mm) AS std_bill_length_mm FROM df\""
+    "All of these queries will cost the same budget in Smartnoise-SQL. The reason is that the smartnoise-sql translates the input query in sub queries, find the answer for each sub query for the budget in input and then assemble the results. For the first 'standard deviation' query, it requires a count, an average, and only then the computation for the standard deviation. Hence, to save budget it is better to make a general query directly and retrieve all the 'sub-answers'."
```

Comment:
findS the answer for each sub-query for the budget given as input and then assembleS the results.